include/linker/ram-nocache.ld

```diff
@@ -0,0 +1,21 @@
+/* Non-cached region of RAM */
+SECTION_PROLOGUE(_NOCACHE_SECTION_NAME,(NOLOAD),)
+{
+#ifdef CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
+	. = ALIGN( 1 << LOG2CEIL(_nocache_ram_end - _nocache_ram_start));
+#endif
+#ifdef CPU_HAS_ARM_MPU
```

Comment:
It is not, really, CPU_HAS_ARM_MPU, rather, ARM_MPU, because HW capabilities do not imply that we build Zephyr with ARM MPU support.

Reply:
**Update:**

So: if CPU_HAS_ARM_MPU and ARM_MPU are not defined, then the CONFIG_NOCACHE_MEMORY option is never defined. So, does it make sense to query these variables inside this script, if the script is only included if CONFIG_NOCACHE_MEMORY is enabled?

Reply:
The idea with this linker script was to make it generic, even if ARM is the only supported architecture for now. I can also move it to an ARM specific directory, and it can be moved back to the top include directory once support for more architectures is added.

Reply:
I believe it is better to move it to arm subdirectory; it is safer and less confusing.
We should not have ARM specific options in a generic linker script overlay.

Still - I believe you can remove the CPU_HAS_ARM_MPU - since this is a prerequisite for the inclusion of this overlay.